packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/ModelReaderWriterContextDefinition.cs

```diff
@@ -310,7 +344,7 @@ private static bool ImplementsModelReaderWriter(TypeProvider typeProvider)
         }
 
         private static void AddAttributeForType(
-            List<MethodBodyStatement> attributes,
+            Dictionary<string, MethodBodyStatement> attributes,
```

Comment:
Still using a dictionary here for deduping. Once we have the sets of framework types and generated types can we just add all of them to a new HashSet that uses a type comparer for deduping?

Reply:
Can't just put them into the same HashSet<CSharpType>, because TypeProvider is needed for https://github.com/microsoft/typespec/blob/07e6593158161fb887cf60a18579d4ec75257c52/packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/ModelReaderWriterContextDefinition.cs#L346-L354
So, once you have both CSharpType/Type(always framework type) and TypeProvider, you can't put them in the same HashSet.